src/nccl_ofi_rdma.c

```diff
@@ -3109,60 +3062,31 @@ static int recv(nccl_net_ofi_recv_comm_t *recv_comm, int n, void **buffers,
 	nccl_net_ofi_rdma_req_t *req = NULL;
 	nccl_net_ofi_rdma_recv_comm_t *r_comm = (nccl_net_ofi_rdma_recv_comm_t *)recv_comm;
 	nccl_net_ofi_rdma_mr_handle_t **mr_handles = (nccl_net_ofi_rdma_mr_handle_t **)mhandles;
-	int dev_id = r_comm->base.base.dev_id;
-
-	/* Retrieve and validate endpoint */
-	nccl_net_ofi_rdma_ep_t * ep =
-		(nccl_net_ofi_rdma_ep_t *)r_comm->base.base.ep;
-	if (OFI_UNLIKELY(ep == NULL)) {
-		ret = -EINVAL;
-		NCCL_OFI_WARN("Invalid endpoint provided");
-		goto error;
-	}
-
-	ret = process_cq_if_pending(ep);
-	if (ret == -EAGAIN) {
-		/* Network is still busy. Return NULL to NCCL. */
-		*base_req = NULL;
-		ret = 0;
-		goto error;
-	} else if (ret != 0) {
-		goto error;
-	}
 
-	/* Retrieve and validate device */
-	nccl_net_ofi_rdma_device_t *device =
-		(nccl_net_ofi_rdma_device_t*)ep->base.device;
-	if (OFI_UNLIKELY(device == NULL)) {
-		ret = -EINVAL;
-		NCCL_OFI_WARN("Invalid device provided");
-		goto exit;
-	}
+	assert(r_comm != NULL);
 
-	/* Support only NCCL_OFI_MAX_REQUESTS inflight reqs. */
 	if (OFI_UNLIKELY(r_comm->num_inflight_reqs == NCCL_OFI_MAX_REQUESTS)) {
 		ret = -ENOSPC;
 		NCCL_OFI_WARN("Can not support more than %d inflight requests",
 			      NCCL_OFI_MAX_REQUESTS);
 		goto error;
 	}
 
-	/* Currently, plugin doesn't support grouped receives */
-	if (n > NCCL_OFI_MAX_RECVS) {
-		NCCL_OFI_WARN("Group receives are not supported!");
-		ret = -ENOTSUP;
-		goto error;
-	}
+	int dev_id = r_comm->base.base.dev_id;
 
-	if (OFI_UNLIKELY(mr_handles == NULL)) {
```

Comment:
It seems this check was lost. Should we put it in api.c?

Reply:
fixed